### PR TITLE
[FEATURE] Ability to Pass Authentication Parameters Individually

### PR DESCRIPTION
**Pull request by anatbal**
**As mentioned in #39:**
Allow login with environment variables instead of json metadata file: passing in the Subscription ID, Tenant ID, Client ID, and Client Secret as individual parameters instead of bundled in a single JSON object (`creds` today). 
This would allow for easier use in pipelines that use Terraform (as it requires them split out). It would also make secret rotation easier as you don't have to rebuilt the JSON object.
 ```yaml
 - uses: azure/login@v1.1
       with:
         creds_client_id: ${{ secrets.CLIENT_ID }}
         creds_client_secret: ${{ secrets.CLIENT_SECRET }}
         creds_subscription_id: ${{ secrets.SUBSCRIPTION_ID }}
         creds_tenant_id: ${{ secrets.TENANTID }}
 ```

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -39,13 +39,13 @@ async function main() {
...
-        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let creds = core.getInput('creds', { required: false });
+        let secrets = creds ? new SecretParser(creds, FormatType.JSON) : null;
+        let servicePrincipalId = core.getInput('creds_client_id', { required: false }) ? core.getInput('creds_client_id', { required: false}) : secrets.getSecret("$.clientId", false);
+        let servicePrincipalKey = core.getInput('creds_client_secret', { required: false }) ? core.getInput('creds_client_secret', { required: false}) : secrets.getSecret("$.clientSecret", false);
+        let tenantId = core.getInput('creds_tenant_id', { required: false }) ? core.getInput('creds_tenant_id', { required: false }) : secrets.getSecret("$.tenantId", false);
+        let subscriptionId = core.getInput('creds_subscription_id', { required: false }) ? core.getInput('creds_subscription_id', { required: false }) : secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = creds ? secrets.getSecret("$.resourceManagerEndpointUrl", false) : "https://management.azure.com/";
```
Comment (BALAGA-GAYATRI) [inline src/main.ts:48]: @anatbal resourceManagerEndpointUrl is different for sovereign clouds for example us gov-  "resourceManagerEndpointUrl": "https://management.usgovcloudapi.net/", we can't default it to "https://management.azure.com/". Ideally this should also be passed separately in that case.  @N-Usha ,@kmkumaran any workaround for this other than explicitly passing resourceManagerEndpoint.

Comment (anatbal) [reply src/main.ts]: @BALAGA-GAYATRI, I changed the resourceManagerEndpointUrl to also be passed separately and fixed the default value. Let me know if that looks okay to you.

File: action.yml [hunk 2]
```diff
@@ -4,7 +4,19 @@ description: 'Authenticate to Azure and run your Az CLI or Az
 inputs:
```
Comment (BALAGA-GAYATRI) [inline action.yml:4]: @anatbal creds_resource_manager_endpoint_url should also be added to input fields

Comment (anatbal) [reply action.yml:4]: Thanks @BALAGA-GAYATRI , I fixed it

**Comment by ericsampson** [comment]
bump for re-review @BALAGA-GAYATRI ? :)

**Comment by kaverma** [comment]
@BALAGA-GAYATRI , is this ready to be merged?

**Comment by BALAGA-GAYATRI** [comment]
Hey @anatbal ! Thank you for the contribution towards this action. We have added OIDC login mechanism to the action which now accepts individual parameters as inputs. Please refer [sample workflow](https://github.com/Azure/login#sample-workflow-that-uses-azure-login-action-using-oidc-to-run-az-cli-linux) and [configure OIDC](https://github.com/Azure/login#configure-a-service-principal-with-a-federated-credential-to-use-oidc-based-authentication) for more details. So closing this PR.

**Pull request closed without merging**